include/pokeathlon/pokeathlon.h

```diff
@@ -0,0 +1,91 @@
+#ifndef POKEHEARTGOLD_POKEATHLON_H
+#define POKEHEARTGOLD_POKEATHLON_H
+
+#include "overlay_manager.h"
+
+// Forward declarations
+typedef struct PokeathlonCourseWork PokeathlonCourseWork;
+typedef struct PokeathlonCourseArgs PokeathlonCourseArgs;
+
+// Pokeathlon course arguments structure
+struct PokeathlonCourseArgs {
+    u8 filler_0[4]; // 0x000
+    u32 field_4;    // 0x004 - Mode flag (1 = special mode)
```

Comment:
eventually you'll want to rename this, though I understand if you don't want to right now